python_modules/libraries/dagster-components/dagster_components/impls/dbt_project.py

```diff
@@ -20,9 +21,16 @@ class DbtProjectParams(BaseModel):
 
 
 class DbtGenerateParams(BaseModel):
-    init: bool = False
+    init: bool = Field(default=False)
     project_path: Optional[str] = None
 
+    @staticmethod
+    @click.command
+    @click.option("--project-path", "-p", type=click.Path(resolve_path=True), default=None)
+    @click.option("--init", "-i", is_flag=True, default=False)
+    def cli(project_path: Optional[str], init: bool) -> "DbtGenerateParams":
+        return DbtGenerateParams(project_path=project_path, init=init)
+
```

Comment:
This is the sort of thing that could in theory be automatically generated (and libraries exist to do such things), but it's easy enough to hand-roll for now